### PR TITLE
M5.3: outcome/override evidence report + M-plan status analysis

### PR DESCRIPTION
**Pull request by Szczepanov**
## Summary

- Adds a dated point-in-time status check for the `M` capability plan (Multidomain session authoring, execution and evidence), confirming the 2026-08-19 cutline chain (M3.7 → bounded M3.8 → M4.3 → M5.1 → M5.2) was fully complete and identifying **M5.3** as the sole unblocked next item.
- Implements **M5.3 (outcome/override evidence report)**, scoped exactly per the plan's own 2026-08-19 cutline: a deterministic report/export over already-recorded evidence, *not* a dedicated history dashboard (`components/session/ResponseHistory.tsx` stays unbuilt until a usage trigger fires).
  - `responses/outcome.ts` — `deriveSessionOutcome`: a pure, versioned `passed | caution | reactive | unknown` summary derived from M5.1/M5.2's `SessionResponse` facts plus the canonical check-in's linked `RegionTissueResponse` values, reusing `engine/injuryPolicy.ts`'s existing tissue-severity mapping rather than restating a second threshold table. `unknown` covers both zero data and immediate-only data with no follow-up yet — never a fabricated `passed`.
  - `responses/outcomeReport.ts` — `buildSessionOutcomeReport` + `sessionOutcomeReportToCsv`: the deterministic, injury-prediction-free report/export surface.
  - `services/sessionOutcomeReportService.ts` — composes existing, already-tested services (`sessionExecutionService`, `sessionResponseService`, `checkinService`, `resolveSessionDefinition`, `comparePlannedVsPerformed`) into the report over a date range. Constructor-injected dependencies follow `StrengthHistoryReadService`'s established pattern for emulator-free unit testing.
  - `sessions/architecture.test.ts` — extends the M0.3 dependency boundary so no selection/optimizer module can import the evidence-only outcome function (D-MPOLICY).
  - **Scoping note:** the plan asks to "record athlete override reason," reusing `SessionAdjustment.athleteReason`. No UI in the repo captures a structured reason for *any* session today (that field has no writer anywhere), so rather than add a second unused enum field, `overrideReason` is threaded through `deriveSessionOutcome`'s input as a plain optional, typed parameter — the record shape exists for a future capture point, but nothing here fabricates a value. `override.note` is populated automatically from data that already exists.
- No `firestore.rules` change — nothing new is persisted.
- Plan docs (`docs/plans/multidomain-session-authoring-execution-and-evidence.md`, `docs/plans/README.md`) updated: M5.3 marked complete with an outcome writeup, and the README's M-plan summary/table row brought back in sync with the plan's own already-recorded completions.

## Validation

Results:
- `npm run check` (typecheck, lint, full unit suite, catalog validation): pass — 1760 unit tests passed, 0 failed (up from 1700 before this change), no typecheck/lint errors.
- `npm run test:rules` (Firestore emulator rules suite): pass — 83/83, confirming no rules regressions (expected, since no schema/rules were touched).
- 31 new unit tests added across `responses/outcome.test.ts`, `responses/outcomeReport.test.ts`, `services/sessionOutcomeReportService.test.ts`, covering status derivation (all four states + edge cases), CSV/report determinism and escaping, and service composition/exclusion behavior (in-progress executions excluded, tissue attribution scoped to the correct source session, comparison only resolved for completed executions).

- [x] `cd app && npm run check` (frontend changes)
- [x] `cd app && npm run test:rules` (Firestore rules changes) — run to confirm no regressions, though no rules were changed
- [ ] `uv run pytest` (backend changes) — not applicable, no backend changes
- [ ] `uv run ruff check .` and `uv run mypy src/garmin_sync` (backend changes) — not applicable
- [ ] `cd app && npm run simulate:scenarios` (recommendation-engine changes) — not applicable; no engine decision logic touched, and the architecture test enforces that this evidence-only module cannot be imported by any selection/optimizer module
- [ ] `cd app && npm run visual:refresh` (material UI changes) — not applicable; no UI was added (deliberately, per the M5.3 cutline)

## Risk and reviewer guidance

Low-to-moderate risk: purely additive (three new modules, one architecture-test extension, doc updates), no existing runtime path is modified, and nothing new is persisted to Firestore. The main thing worth a close look is the scoping decision on `overrideReason` (see Summary) — reviewers should confirm that not adding a new persisted "reason" field, and instead threading an unpopulated-for-now optional parameter, is the right call versus extending the M5.1 `SessionResponse` schema now.

## Domain invariants

- [x] Firestore writes remain user-scoped under `users/{APP_USER_ID}/...`; no `default_user` or top-level recovery snapshots were introduced — not applicable, this change introduces no new writes at all.
- [x] Calendar-date logic uses `Europe/Warsaw`; completed `totalSteps` still means the previous calendar day (`D - 1`) — not applicable, no date/step logic touched.
- [x] No credentials, tokens, service-account files, or raw health payloads are included.
- [x] Recommendation decision logic changed: `POLICY_VERSION` was evaluated and relevant architecture/ADR documentation was updated — not applicable; this is evidence-only and cannot reach recommendation/selection logic (enforced by the new architecture test), so no `POLICY_VERSION` bump is needed.

## Screenshots or recordings

Not applicable — no UI change; the M5.3 cutline explicitly calls for report/export only, not a UI surface.

---
🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added evidence-based session outcome reporting with `passed`, `caution`, `reactive`, and `unknown` statuses.
  * Included follow-up details, override notes, source evidence, and planned-versus-performed comparisons.
  * Added chronological report generation and deterministic CSV export for completed sessions.
  * Excluded in-progress sessions and preserved unknown outcomes when required information is unavailable.

* **Documentation**
  * Updated session planning documentation to reflect the completed outcome reporting milestone and CSV export capability.

### CODE REVIEW

File: app/src/responses/outcome.ts [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+/** Prefers a later_day/next_morning note over an immediate one (today's `SessionResponse`
+ * production callers never write `immediate` anyway -- see `responses/models.ts`), and the
+ * most recently recorded note when more than one window carries one. */
+function mostRelevantNote(responses: readonly SessionResponse[]): string | undefined {
+    const withNotes = responses.filter(response => response.note || response.techniqueNote);
+    if (withNotes.length === 0) return undefined;
+    const sorted = [...withNotes].sort((a, b) => b.updatedAt.localeCompare(a.updatedAt));
+    return sorted[0]?.note ?? sorted[0]?.techniqueNote;
+}
```
[the remark this answers is not in the document]
Comment (Szczepanov) [reply app/src/responses/outcome.ts:122]: Fixed in 758aa3d — `mostRelevantNote` now prefers a `later_day`/`next_morning` note over a more-recently-updated `immediate` one, with a regression test.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: app/src/responses/outcomeReport.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+const CSV_COLUMNS: (keyof SessionOutcomeReportRow)[] = [
+    'date', 'sourceKind', 'sourceId', 'status', 'hasFollowUpData', 'responseCount',
+    'tissueRegions', 'overrideReason', 'note', 'completedStepsCount',
+    'missingRequiredStepsCount', 'totalPlannedSteps', 'policyVersion',
+];
+
+function csvField(value: string | number | boolean): string {
+    const text = String(value);
+    if (/[",\n]/.test(text)) return `"${text.replaceAll('"', '""')}"`;
+    return text;
```
[the remark this answers is not in the document]
Comment (Szczepanov) [reply app/src/responses/outcomeReport.ts:64]: Fixed in 758aa3d — quoting condition now matches `\r` as well as `\n`, with a regression test.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: app/src/services/sessionOutcomeReportService.ts [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+            const checkinDates = new Set(responses.map(response => response.checkinRef.date));
+            const tissueResponses: RegionTissueResponse[] = [];
+            for (const date of checkinDates) {
+                const regions = await tissueResponsesForDate(date);
+                for (const region of regions) {
+                    if (region.sourceSessionRef?.kind === 'execution' && region.sourceSessionRef.id === execution.executionId) {
+                        tissueResponses.push(region);
+                    }
+                }
+            }
```
[the remark this answers is not in the document]
Comment (Szczepanov) [reply app/src/services/sessionOutcomeReportService.ts]: Good catch — this was a real gap. Fixed in 758aa3d: tissue-evidence discovery now scans every check-in in the report range (plus a bounded lookahead buffer) directly for a matching `sourceSessionRef`, instead of being keyed off which dates have a `SessionResponse`. Added a regression test covering a tissue reaction with no `SessionResponse` at all.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: app/src/sessions/architecture.test.ts [hunk 4]
```diff
@@ -144,4 +144,29 @@ describe('Multidomain sessions architecture and dependency boundaries (M0.3 / AD
...
+
+        for (const [mod, imports] of graph.entries()) {
+            const importsOutcome = imports.some(imp => imp === 'responses/outcome.ts');
+            if (!importsOutcome) continue;
+            for (const forbidden of forbiddenSelection) {
+                expect(
+                    mod.startsWith(forbidden),
+                    `Selection module "${mod}" must not import evidence-only "responses/outcome.ts"`,
+                ).toBe(false);
+            }
```
[the remark this answers is not in the document]
Comment (Szczepanov) [reply app/src/sessions/architecture.test.ts:191]: Fixed in 758aa3d — the check now walks full reverse reachability from `responses/outcome.ts` (BFS over the reverse import graph) rather than only direct edges, so a transitive path through an intermediate module is caught too.

---
_Generated by [Claude Code](https://claude.ai/code)_
